### PR TITLE
IO fix

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

**Comment by bgavrilMS** [comment]
PR build: https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=906758&view=results

File: src/Microsoft.Identity.Client.Extensions.Msal/Accessors/FileWithPermissions.cs [hunk 1]
```diff
@@ -62,14 +64,16 @@ public static void WriteToNewFileWithOwnerRWPermissions(string path, byte[] data
...
+            {
+            }
 
             // Setting permissions to 0600 
             const int _0600 = S_IRUSR | S_IWUSR; // read & write only for user
             int result = chmod(filePath, _0600);
 
             if (result != 0)
-            {
+            {
```
Comment (jmprieur) [inline src/Microsoft.Identity.Client.Extensions.Msal/Accessors/FileWithPermissions.cs:76]: extra space?

File: src/Microsoft.Identity.Client.Extensions.Msal/Accessors/FileWithPermissions.cs [hunk 2]
```diff
@@ -62,14 +64,16 @@ public static void WriteToNewFileWithOwnerRWPermissions(string path, byte[] data
         /// </summary>
         private static void WriteToNewFileWithOwnerRWPermissionsUnix(string filePath, byte[] data)
         {
-            File.Create(filePath);
+            using (File.Create(filePath))
+            {
```
Comment (jmprieur) [inline src/Microsoft.Identity.Client.Extensions.Msal/Accessors/FileWithPermissions.cs:68]: Why is it? Is it to be sure the file is indeed created?
Do you need to create the folder before?

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @bgavrilMS

File: src/Microsoft.Identity.Client.Extensions.Msal/Accessors/FileWithPermissions.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/Microsoft.Identity.Client.Extensions.Msal/Accessors/FileWithPermissions.cs:68]: File.Create returns a stream for writing.

**Review by pmaytak — COMMENTED** [review]
Any idea why this fixed the Mac and Linux tests?

**Comment by bgavrilMS** [comment]
> Any idea why this fixed the Mac and Linux tests?

Yes, because `File.Open` actually returns a FileStream, i.e. keeps the file open. Looks like on Windows the GC closes the stream quickly, but not on Mac / Linux. After applying the permissions, I re-open the file to write to it (in Append mode)

**Pull request merged**